parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -222,6 +222,24 @@ private static MessageType getParquetTypeWithIds(
     return ParquetSchemaUtil.addFallbackIds(type);
   }
 
+  /**
+   * Returns a list of offsets in ascending order determined by the starting position of the row
+   * groups.
+   */
+  public static List<Long> getSplitOffsets(InputFile file) {
+    try (ParquetFileReader reader = ParquetFileReader.open(ParquetIO.file(file))) {
+      ParquetMetadata md = reader.getFooter();
+      List<Long> splitOffsets = Lists.newArrayListWithExpectedSize(md.getBlocks().size());
+      for (BlockMetaData blockMetaData : md.getBlocks()) {
+        splitOffsets.add(blockMetaData.getStartingPos());
```

Comment:
Shall we just call the overloaded method once we have `ParquetMetadata`?

```
return getSplitOffsets(reader.getFooter());
```

Reply:
We could do that. I'll change the PR.

Reply:
Fixed in latest commit.